tests/conftest.py

```diff
@@ -4523,3 +4523,19 @@ def finalizer():
 
     request.addfinalizer(finalizer)
     return factory
+
+
+@pytest.fixture(scope="function", autouse=True)
+def collect_mg_rules():
+    config.REPORTING["ocs_must_gather_collection"] = True
+    config.REPORTING["ocp_must_gather_collection"] = False
+
+
+@pytest.fixture()
+def disable_ocs_mg_collection_on_failure():
```

Comment:
If I understand it correctly, this approach will change the configuration value and it will be changed also for all the following tests (until one of those fixtures are not used again in another test) - is my understanding correct? --> correct and I added a new fixture
```
@pytest.fixture(scope="function", autouse=True)
def collect_mg_rules():
    config.REPORTING["ocs_must_gather_collection"] = True
```

Reply:
I meant it generally for all the new fixtures - when they are used, they change the configuration for all following tests until another fixture is explicitly used in some test.

For example if I'll have following tests:

1. `test_a(disable_ocs_mg_collection_on_failure)`
1. `test_b()`
1. `test_c(collect_mg_rules)`
1. `test_d()`

* Then if I'll run them all and if they will be executed in this order, then `test_b` will have `ocs_must_gather_collection` set to `False`, because it will be set by the fixture in `test_a`. Then `test_c` will change it back to `True`.
* If I'll run only `test_a`, `test_b` and `test_d`, then both `test_b` and `test_d` will have the `ocs_must_gather_collection` set to `False`, even though they didn't required it.
* And if I'll run for example only `test_b`, `test_c` and `test_d`, all the tests will have `ocs_must_gather_collection` set to `True`.

This is what I meant when I wrote above, that one test changes behavior of the following tests., which is not good.

I think, that the approach should be, that if you want to change the defaults for one test, you should change it in the beginning of the test, but save the previous values and then restore the previous values once the test finish.